remo/domain/dataset.py

```diff
@@ -120,25 +120,45 @@ def annotation_sets(self) -> List[AnnotationSet]:
 
     def add_annotations(self, annotations: List[Annotation], annotation_set_id: int = None):
         """
-        Faster upload of annotations to the Dataset via file conversion.
+        Fast upload of annotations to the Dataset.
         
-        If there are no Annotation Sets, an Annotation Set is automatically created.
-        If annotation_set_id is not specified, annotations are added to the default Annotation Set.
-        If the default Annotation Set's task doesn't match the annotations task, a new Annotation Set is also created.
+        An Annotation Set will be automatically created and populated in these cases:
+            - there are no Annotation Sets
+            - annotation_set_id = -1
+            - the default Annotation Set's task doesn't match the annotation task of the annotations
+
+        Otherwise, annotations will be added to the relevant Annotation Set (default or as specified by annotation_set_id)
         
         Args:
             annotations: list of Annotation objects
-            (optional) annotation_set_id: annotation set id
+            (optional) annotation_set_id: annotation set id. If annotation_set_id = -1, a new annotation set will be created
+            
+        Example::
+            urls = ['https://remo-scripts.s3-eu-west-1.amazonaws.com/open_images_sample_dataset.zip']
+            my_dataset = remo.create_dataset(name = 'D1', urls = urls)
+            images = my_dataset.images()
+            my_image = images[1]
+            annotations = []
+
+            annotation = remo.Annotation()
+            annotation.img_filename = my_image.name
+            annotation.classes='Human hand'
+            annotation.bbox=[227, 284, 678, 674]
+            annotations.append(annotation)
+
+            annotation = remo.Annotation()
+            annotation.img_filename = my_image.name
+            annotation.classes='Fashion accessory'
+            annotation.bbox=[496, 322, 544,370]
+            annotations.append(annotation)
+
+            my_dataset.add_annotations(annotations)
         """
         annotation_set = self.get_annotation_set(annotation_set_id)
         temp_path, list_of_classes = create_tempfile(annotations)
         
-        if not annotation_set:
-
-            self.create_annotation_set(annotation_task=annotations[0].task, name='my_ann_set',
-                                       classes = list_of_classes, path_to_annotation_file = temp_path)
         
-        elif annotation_set.task != annotations[0].task:
+        if (not annotation_set) or (annotation_set.task != annotations[0].task) or (annotation_set_id==-1):
```

Comment:
when `annotation_set_id == -1`?

Reply:
see below

Reply:
`(optional) annotation_set_id: annotation set id. If annotation_set_id = -1, a new annotation set will be created`

I don't think it's needed, `annotation_set_id` can be just None

if you want to force creating annotation set, I guess would be better to add boolean flag like `create_new`